src/cluster/sharding.rst

```diff
@@ -12,290 +12,490 @@
+as in step 2.
+
+Verify that the node is in not maintenance mode by performing a ``GET
+/_up`` on that node's individual endpoint.
+
+Finally, check that your load balancer has returned the node to the pool
+of available backend nodes.
 
-Moving Shards
-=============
+Update cluster metadata again to remove the source shard
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Add, then delete
-----------------
+Now, remove the source shard from the shard map the same way that you
+added the new target shard to the shard map in step 2. Be sure to add
+the ``["remove", <range>, <source-shard>]`` entry to the end of the
+changelog as well as modifying both the by_node and by_range sections of
+the database metadata document.
 
-In the world of CouchDB there is no such thing as "moving" shards, only adding
-and removing shard replicas.
-You can add a new replica of a shard and then remove the old replica,
-thereby creating the illusion of moving.
-If you do this for a database that has ``n=1``,
-you might be caught by the following mistake:
+Remove the shard and secondary index files from the source node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-#. Copy the shard onto a new node.
-#. Update the metadata to use the new node.
-#. Delete the shard on the old node.
-#. Oh, no!: You have lost all writes made between 1 and 2.
+Finally, you can remove the source shard by deleting its file from the
+command line on the source host, along with any view shards:
 
-To avoid this mistake, you always want to make sure
-that both shards have been live for some time
-and that the shard on your new node is fully caught up
-before removing a shard on an old node.
-Since "moving" is a more conceptually (if not technically)
-accurate description of what you want to do,
-we'll use that word in this documentation as well.
+.. code::bash
 
-Moving
-------
+    $ rm <couch-dir>/data/shards/<range>/<dbname>.<datecode>.couch
+    $ rm -r <couch-dir>/data/.shards/<range>/<dbname>.<datecode>*
 
-When you get to ``n=3`` you should start moving the shards instead of adding
-more replicas.
+Congratulations! You have moved a database shard. By adding and removing
+database shards in this way, you can change the cluster's shard layout,
+also known as a shard map.
 
-We will stop on ``n=2`` to keep things simple. Start node number 3 and add it to
-the cluster. Then create the directories for the shard on node3:
+Specifying database placement
+-----------------------------
 
-.. code-block:: bash
+Database shards can be configured to live solely on specific nodes using
+placement rules.
 
-    mkdir -p data/shards/00000000-7fffffff
+First, each node must be labeled with a zone attribute. This defines
+which zone each node is in. You do this by editing the node’s document
+in the ``/nodes`` database, which is accessed through the “back-door”
+(5986. port. Add a key value pair of the form:
```

Comment:
... through the "backdoor" port (5986).

Reply:
Is there an official name for this port? It's called "node's private port" above. Can we be consistent in what we call this?  "Local administration port" or similar?

Reply:
"node-local port" and "clustered port" are the most common terms.